### PR TITLE
Show app located path while running the occ app: list command

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Show app located path while running the occ app: list command  

## Related Issue

- Fixes https://github.com/owncloud/enterprise/issues/4012

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by JanAckermann** [comment]
TBD: Raise documentation ticket 
https://doc.owncloud.com/server/admin_manual/configuration/server/occ_command.html#app-commands

**Review by phil-davis — COMMENTED** [review]
For the existing command:
`occ app:list`

The default output is just a list of apps, for example:
```
Enabled:
  - comments: 0.3.0
  - dav: 0.6.0
  - federatedfilesharing: 0.5.0
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_sharing: 0.14.0
  - files_texteditor: 2.3.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - testing: 0.1.0
  - updatenotification: 0.2.1
Disabled:
  - activity
  - admin_audit
...
```
But with this PR so far:
```
Enabled:
  - comments:
    - Version: 0.3.0
  - dav:
    - Version: 0.6.0
  - federatedfilesharing:
    - Version: 0.5.0
  - federation:
    - Version: 0.1.0
  - files:
    - Version: 1.5.2
  - files_external:
    - Version: 0.7.1
  - files_sharing:
    - Version: 0.14.0
  - files_texteditor:
    - Version: 2.3.0
  - files_trashbin:
    - Version: 0.9.1
  - files_versions:
    - Version: 1.3.0
  - provisioning_api:
    - Version: 0.5.0
  - systemtags:
    - Version: 0.3.0
  - testing:
    - Version: 0.1.0
  - updatenotification:
    - Version: 0.2.1
Disabled:
  - activity:
    - Version
  - admin_audit:
    - Version
```
For apps where the version is not known, `Version` is shown with nothing after it.
These are apps that exist in `apps_paths` but are disabled.
I wonder what we should do about that? Not report `Version` at all when the version information is not available?

**Comment by JanAckermann** [comment]
@phil-davis can you explain why the disabled apps dont show version tags?

**Comment by phil-davis** [comment]
> can you explain why the disabled apps dont show version tags?

I think that the version information is the version that gets stored in the database when an app is enabled. If there is no database entry, then the code does not run off an try to directly open `appinfo/info.xml` to see what the version would be if the administrator does enable the app.

Of the top of my head, I don't remember why we did not make the `occ` command do that extra work.

**Comment by JanAckermann** [comment]
> TBD: Raise documentation ticket
> https://doc.owncloud.com/server/admin_manual/configuration/server/occ_command.html#app-commands

Not needed due, this is now a default behaviour

File: core/Command/App/ListApps.php [hunk 1]
```diff
@@ -112,14 +112,27 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
 			foreach ($enabledApps as $app) {
-				$apps['enabled'][$app] = (isset($versions[$app])) ? $versions[$app] : true;
+				$appDetailRecord = [];
+
+				if (isset($versions[$app])) {
+					$appDetailRecord['Version'] = $versions[$app];
+				}
+
+				$appDetailRecord['Path'] = \OC_App::getAppPath($app);
+				$apps['enabled'][$app] =$appDetailRecord;
```
Comment (phil-davis) [inline core/Command/App/ListApps.php:122]:
```suggestion
				$apps['enabled'][$app] = $appDetailRecord;
```
That is odd that code-style does not enforce consistent spaces around `=`

**Review by phil-davis — COMMENTED** [review]
LGTM. Nice to see the test changes that cover this.

**Review by JammingBen — APPROVED** [review]
Works and looks good to me :+1:

**Pull request merged**